geoh5py/groups/interpretation_section.py

```diff
@@ -0,0 +1,411 @@
+
+            self._interpretation_curves = [
+                self._verify_object(curve, "Curve") for curve in curves
+            ]
+
+            self.add_children(self._interpretation_curves)  # type: ignore
+
+        self._update_to_metadata("Interpretation curves", self._interpretation_curves)
+
+    @property
+    def interpretation_sections(self) -> list[InterpretationSectionParams] | None:
+        """
+        Get the interpretation sections.
+
+        :return: The interpretation sections.
+        """
+        if self._interpretation_sections is None:
+            self.interpretation_sections = self._load_from_metadata(
+                "Interpretation sections"
+            )
+
+        return self._interpretation_sections
+
+    @interpretation_sections.setter
+    def interpretation_sections(
+        self, sections: list[InterpretationSectionParams] | None
+    ):
+        """
+        Set the interpretation sections.
+
+        :param sections: The interpretation sections.
+        """
+        if sections is None:
+            self._interpretation_sections = None
+        else:
+            if not isinstance(sections, list):
+                raise TypeError(
+                    "Interpretation sections must be a list of InterpretationSectionParams."
+                )
+
+            self._interpretation_sections = [
+                InterpretationSectionParams.verify(section) for section in sections
+            ]
+
+        self._update_to_metadata(
+            "Interpretation sections", self._interpretation_sections
+        )
+
+    def remove_interpretation_curve(self, curves: Curve | list[Curve]):
+        """
+        Remove an interpretation curve.
+
+        :param curves: The interpretation curve to remove.
+        """
+        self._load_attributes()
+        if self._interpretation_curves is None:
+            return
+
+        if not isinstance(curves, list):
+            curves = [curves]
+
+        for curve in curves:
+            curve = self._verify_object(curve, "Curve")
+            if curve in self._interpretation_curves:
+                self._interpretation_curves.remove(curve)
+
+        self._update_to_metadata("Interpretation curves", self._interpretation_curves)
+
+    def remove_interpretation_section(
+        self, sections: InterpretationSectionParams | list[InterpretationSectionParams]
+    ):
+        """
+        Remove an interpretation section.
+
+        :param sections: The interpretation section to remove.
+        """
+        self._load_attributes()
+        if self._interpretation_sections is None:
+            return
+
+        if not isinstance(sections, list):
+            sections = [sections]
+
+        for section in sections:
+            section = InterpretationSectionParams.verify(section)
+            idx = section.in_list(self._interpretation_sections)
+            if idx is not None:
+                del self._interpretation_sections[idx]
```

Comment:
```suggestion
        new_list = []
        for section in self._interpretation_sections:
            if section not in sections:
                new_list = section
        
        self._update_to_metadata(
            "Interpretation sections", new_list
        )  
```

Also need to cut out the curves that belong to that section.